### DISCUSSION TITLE
Add Tape into test_network_k8s as sample

### DISCUSSION THREAD

**Discussion by SamYuan1990** [hyperledger/fabric-samples#551] [Ideas]
Hi Fabric_samples,

I am planning to add [tape]( https://github.com/Hyperledger-TWGC/tape) in to test network k8s sample.

Please let me know your points.

Thanks and regards
Sam

**Comment by mbwhite** [comment]
Hi @SamYuan1990 , firstly thanks for your support to Fabric!

I think that for the k8s test network, we shouldn't add in too much extra tooling; the idea to try and keep it as easy to extract techniques from.. i.e. keep using it as an example/sample.

However, including something, such as Tape, in something like the test_network_k8s is interesting.  Perhaps it should live else where?  (github.com/hyperledgendary perhaps?)

**Reply by SamYuan1990** [reply]
Hi @mbwhite , I have an idea learned by test_network_k8s. That if we take k8s as infrastructure, follow PSWG paper, adding monitoring and open tracing with performance testing tools there...
![image](https://user-images.githubusercontent.com/7820992/145023820-bda99621-3855-4ce6-8c83-7cecd46cfd20.png)

Which means it sounds like we will have a sandbox for blockchain performance research.
I will explain detail and show a poc demo at:
https://wiki.hyperledger.org/display/PSWG/PSWG+December+14%2C+2021

I suppose if we are going to have a sandbox project, it's better to one of hyperledger labs maybe?

**Reply by SamYuan1990** [reply]
The reason I am asking, is that I see https://github.com/hyperledger/fabric-samples/tree/main/test-network-k8s#areas-for-improvement--todos and https://github.com/hyperledger/caliper/issues/1149, meanwhile, https://github.com/SamYuan1990/distributedtape I had verified that tape at alpha branch supporting for distributed.

**Reply by mbwhite** [reply]
Hyperledger Lab.. sounds a good idea... certainly worth making the Caliper maintainers aware as well. 

An aim of the k8s network, was to try and make it easier to build up and out from the basics.. so we can get to "highervalue" things.. such as what you're suggesting. 

@jkneubuh fyi

**Reply by SamYuan1990** [reply]
yes, I am agree that put the full sandbox into fabric-samples sounds like too heavy. However as performance is one of the top items according to Oct's survey result. I suppose a sandbox can help us know more about performance. which means we may need to create a new lab project from this test-network-k8s?

**Reply by SamYuan1990** [reply]
@mbwhite , I had submit the labs propose as https://github.com/hyperledger-labs/hyperledger-labs.github.io/pull/185, please help take a look at. if you like it, please help with sponsor.

**Reply by SamYuan1990** [reply]
https://github.com/hyperledger-labs/PerformanceSandBox @mbwhite , I suppose we can answer this discussion with lab project PerformanceSandBox.
